src/com/lilithsthrone/game/character/body/OrificeVagina.java

```diff
@@ -290,21 +233,45 @@ public String setPlasticity(GameCharacter owner, int plasticity) {
 		}
 	}
 
-	@Override
-	public boolean isVirgin() {
-		return virgin;
-	}
 
 	@Override
-	public void setVirgin(boolean virgin) {
-		this.virgin = virgin;
-	}
+	public String setIngestionRate(GameCharacter owner, int ingestionRate) {
+		int oldIngestionSpeed = this.ingestionRate;
+		this.ingestionRate = Math.max(0, Math.min(ingestionRate, IngestionRate.FIVE_MAXIMUM.getValue()));
+		int ingestionRateChange = this.ingestionRate - oldIngestionSpeed;
 
-	@Override
-	public boolean hasOrificeModifier(OrificeModifier modifier) {
-		return orificeModifiers.contains(modifier);
+		if(owner==null) {
+			return "";
+		}
+
+		if (ingestionRateChange == 0) {
+			if(owner.isPlayer()) {
+				return "<p style='text-align:center;'>[style.colourDisabled(Your ingestion rate of fluids taken vaginally doesn't change...)]</p>";
+			} else {
+				return UtilText.parse(owner, "<p style='text-align:center;'>[style.colourDisabled([npc.namePos] ingestion rate of fluids taken vaginally doesn't change...)]</p>");
+			}
+		}
+
+		String ingestionDescriptor = getIngestionRate().getDescriptor();
+		if (ingestionRateChange > 0) {
+			return UtilText.parse(owner,
+					"<p>"
+							+ "[npc.Name] [npc.verb(feel)] strange warmth in [npc.her] vagina. "
+							+ "[npc.She] [npc.verb(believe)] it should be more absorbent now.<br/>"
+							+ "Now [npc.her] igestion rate for fluids taken vaginally is [style.boldSex(" + ingestionDescriptor + ")]!"
```

Comment:
same
also you take something orally, vaginally is not ingested/absorbed; at least the descriptions need a better wording. This may apply to other descriptions that I commented to as well.